pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoder.java

```diff
@@ -99,7 +99,7 @@ public GenericRow decode(byte[] payload) {
   }
 
   private Object stringToDataType(FieldSpec spec, String inString) {
-    if (inString == null) {
+    if (inString == null || inString.isEmpty()) {
```

Comment:
Why are we changing empty string into defaultNull? What if the input has a valid empty string?